### DISCUSSION TITLE
Troubleshooting necessary

### DISCUSSION THREAD

**Discussion by kapparop** [linux-surface/linux-surface#930] [Q&A]
I tried the latest versions of Linux Mint 21 Cinnamon and Fedora 36 and installed the surface kernel on a Surface Pro 4 by following the tutorial.

uname -a contains “-surface”, as intended.

However neither distribution was able to use the touchscreen.

There is no information on something called “Wayland” in the tutorial and neither distribution has it in the software centre. I remember, Ubuntu didn’t have that either.
But the tutorial suggests Wayland is not mandatory for the touchscreen, but you won’t get its full potential. I understand that Xorg should be good enough at least to try if the touchscreen works.

Needless to say, Windows or UEFI use the touchscreen normally.

How can this problem be debugged?

**Comment by qzed** [comment]
Do you have IPTSd installed and `iptsd.service` running?

**Reply by kapparop** [reply]
Thanks for your reply. I ran “sudo systemctl enable iptsd” as instructed in the tutorial and it didn’t show any error.

**Reply by NP-chaonay** [reply]
@kapparop what is output of systemctl status iptsd

**Comment by kapparop** [comment]
It says this :
iptsd.service - Intel Precise Touch & Stylus Daemon
     Loaded: loaded (/lib/systemd/system/iptsd.service; enabled; vendor preset: enabled)
     Active: active (running) since Sun 2022-09-18 21:24:50 CEST; 1min 20s ago
       Docs: https://github.com/linux-surface/iptsd
   Main PID: 653 (iptsd)
      Tasks: 1 (limit: 4488)
     Memory: 384.0K
        CPU: 65ms
     CGroup: /system.slice/iptsd.service
             â””â”€653 /usr/bin/iptsd

sept. 18 21:24:50 Surface-Pro-4 systemd[1]: Started Intel Precise Touch & Stylus Daemon.

**Comment by kapparop** [comment]
I tried this: restarted the computer without the keyboard attached. This time the touchscreen worked after a few seconds.

However, there is no wayland in the package manager and no on‐screen keyboard, which means it can’t really be used as a tablet. You have to have the physical keyboard with it.

**Reply by NP-chaonay** [reply]
> no wayland in the package manager 

it should be, have for APT-based pkg manager, using ```apt-cache find wayland ```

**Comment by kapparop** [comment]
OK, i added the virtual keyboard from Accessibility. It has some problems though, but that’s outside of this subject.

**Comment by kapparop** [comment]
I can’t find any Wayland and the virtual keyboard is barely usable. Also the screen gestures are so imprecise. Thanks for your help, but I will stay with Windows.

**Reply by NP-chaonay** [reply]
> I can’t find any Wayland

even you have found wayland pkg, you dont have to using wayland to make thing work, but on wayland thing seem better.

you could try to do on wayland, but at least X11 should give precise touching. btw 

> screen gestures are so imprecise.

idk about this at this moment
